nat/src/main/java/org/hyperledger/besu/nat/kubernetes/KubernetesNatManager.java

```diff
@@ -83,15 +88,7 @@ protected void doStart() throws NatInitializationException {
       final CoreV1Api api = new CoreV1Api();
       // invokes the CoreV1Api client
       final V1Service service =
-          api
-              .listServiceForAllNamespaces(
-                  null, null, null, null, null, null, null, null, null, null)
-              .getItems()
-              .stream()
-              .filter(
-                  v1Service -> v1Service.getMetadata().getName().contains(besuServiceNameFilter))
-              .findFirst()
-              .orElseThrow(() -> new NatInitializationException("Service not found"));
+          api.readNamespacedService(besuServiceName, besuServiceNamespace, null);
```

Comment:
I think service namespace should be an Optional<String> parameter to the manager.  Not having to explicitly specify the namespace is a useful feature IMO, and can prevent breaking existing clustered implementations.  

The presence or absence of the namespace can gate the behavior of finding first vs explicitly specify the target namespace

Reply:
@alex123012 any word on this?